LX/src/main/java/heronarts/lx/LXSnapshotController.java

```diff
@@ -0,0 +1,161 @@
+package heronarts.lx;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import heronarts.lx.osc.OscMessage;
+
+public class LXSnapshotController extends LXComponent {
+
+  private Integer nextShapshotID = 1;
+
+  private final List<LXSnapshot> snapshots = new ArrayList<LXSnapshot>();
+  private final TreeMap<Integer, LXSnapshot> snapshotsDict = new TreeMap<Integer, LXSnapshot>();
+
+  public interface SnapshotListener {
+    public void snapshotAdded(LXSnapshotController controller, LXSnapshot snapshot);
+    public void snapshotRemoved(LXSnapshotController controller, LXSnapshot snapshot);
+  }
+
+  private final List<SnapshotListener> snapshotListeners = new ArrayList<SnapshotListener>();
+
+  public LXSnapshotController(LX lx) {
+    super(lx);
+    this.label.setDescription("Snapshots");
+  }
+
+  private LXSnapshot _createNewSnapshot() {
+    LXSnapshot ss = new LXSnapshot(this.lx, this.nextShapshotID);
+    ss.initialize();
+    this.nextShapshotID++;
+
+    return ss;
+  }
+
+  private LXSnapshot _addSnapshot(LXSnapshot snapshot) {
+    if (snapshot == null) {
+      throw new IllegalArgumentException("Cannot add null snapshot");
+    }
+    this.snapshots.add(snapshot);
+    this.snapshotsDict.put(snapshot.snapshotID, snapshot);
+    for (SnapshotListener snapshotListener : this.snapshotListeners) {
+      snapshotListener.snapshotAdded(this, snapshot);
+    }
+    return snapshot;
+  }
+
+  public LXSnapshot addSnapshot(LXSnapshot snapshot) {
+    return _addSnapshot(snapshot);
+  }
+
+  public LXSnapshot addSnapshot() {
+    LXSnapshot snapshot = _createNewSnapshot();
+    _addSnapshot(snapshot);
+    return snapshot;
+  }
+
+  public void runSnapshot(int snapshotID) {
```

Comment:
I get the intuitive reason for "run", but feel like this would be more consistent as something like loadSnapshot().

Also wonder whether this should be invoked by integer index on the controller object, vs. being a direct "load" method on the LXSnapshot class itself.